sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/clienttelemetry/ClientTelemetry.java

```diff
@@ -239,7 +271,21 @@ public void close() {
             }).subscribeOn(scheduler);
     }
 
+    private void populateAzureVmMetaData(AzureVMMetadata azureVMMetadata) {
+        this.clientTelemetryInfo.setApplicationRegion(azureVMMetadata.getLocation());
+        this.clientTelemetryInfo.setVmId(azureVMMetadata.getVmId());
+        this.clientTelemetryInfo.setHostEnvInfo(azureVMMetadata.getOsType() + "|" + azureVMMetadata.getSku() +
+            "|" + azureVMMetadata.getVmSize() + "|" + azureVMMetadata.getAzEnvironment());
+    }
+
     private void loadAzureVmMetaData() {
+        AzureVMMetadata metadataSnapshot = azureVmMetaDataSingleton.get();
```

Comment:
loadAzureVmMetaData is part of init and not meant to call multiple times , why we used AtomicReference here ?

Reply:
It gets initialized per cosmos client instance - while the VM metadata is identical per machine - no need to retrieve it for each new client.